src/features/defi/providers/idle/components/IdleManager/Overview/IdleOverview.tsx

```diff
@@ -0,0 +1,176 @@
+  const descriptionQuery = useGetAssetDescriptionQuery({ assetId, selectedLocale })
+
+  const walletAddress = useWalletAddress('0x0000000000000000000000000000000000000000')
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!(vaultAddress && api)) return
+        const [opportunity] = await Promise.all([
+          api.findByOpportunityId(
+            toAssetId({ chainId, assetNamespace, assetReference: vaultAddress }),
+          ),
+        ])
+        if (!opportunity) {
+          return toast({
+            position: 'top-right',
+            description: translate('common.somethingWentWrongBody'),
+            title: translate('common.somethingWentWrong'),
+            status: 'error',
+          })
+        }
+        setOpportunity(opportunity)
+
+        const claimableTokens = await opportunity.getClaimableTokens(walletAddress)
+        setClaimableTokens(claimableTokens)
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('IdleOverview error:', error)
+      }
+    })()
+  }, [api, vaultAddress, chainId, toast, translate, walletAddress])
+
+  const additionalParams: Record<string, any> = {}
+
+  useAppSelector(selectorState => {
+    if (claimableTokens && claimableTokens.length > 0) {
+      let rewardAssets: any[] = []
+      claimableTokens.forEach(token => {
+        const rewardAsset = selectAssetById(selectorState, token.assetId)
+        if (rewardAsset) {
+          rewardAssets.push({
+            ...rewardAsset,
+            cryptoBalance: bnOrZero(token.amount).div(`1e+${asset.precision}`).toPrecision(),
+          })
+        }
+      })
+
+      additionalParams.rewardAssets = rewardAssets
+    }
+  })
+
+  const totalClaimableRewards = useMemo(() => {
+    if (!claimableTokens) return bnOrZero(0)
+    return claimableTokens.reduce((totalRewards, token) => {
+      totalRewards = totalRewards.plus(token.amount)
+      return totalRewards
+    }, bnOrZero(0))
+  }, [claimableTokens])
+
+  const claimDisabled = !totalClaimableRewards || totalClaimableRewards.lte(0)
+
+  if (!opportunity) {
+    return (
+      <Center minW='500px' minH='350px'>
+        <CircularProgress />
+      </Center>
+    )
+  }
+
+  const menu: DefiButtonProps[] = [
+    {
+      label: 'common.deposit',
+      icon: <ArrowUpIcon />,
+      action: DefiAction.Deposit,
+    },
+    {
+      label: 'common.withdraw',
+      icon: <ArrowDownIcon />,
+      action: DefiAction.Withdraw,
+    },
+  ]
+
+  if (!opportunity.metadata.cdoAddress) {
+    menu.push({
+      label: 'common.claim',
+      icon: <FaGift />,
+      action: DefiAction.Claim,
+      variant: 'ghost-filled',
+      colorScheme: 'green',
+      isDisabled: claimDisabled,
+      toolTip: translate('defi.modals.overview.noWithdrawals'),
+    })
+  }
+
+  return (
+    <Overview
+      asset={asset}
+      name={`${underlyingToken.name} Vault (${opportunity.version})`}
```

Comment:
translations

Reply:
@samster91 bump

Reply:
This has been taken from yearn implementation:
https://github.com/shapeshift/web/blob/feat_idle_finance/src/features/defi/providers/yearn/components/YearnManager/Overview/YearnOverview.tsx#L97

Do I need to translate the `vault` word?

Reply:
Let's leave it as is, we'll tackle it in https://github.com/shapeshift/web/issues/2434